v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,128 @@
+
+<img src="{{ 'images/Parallel_Statement_Normal_Execution.png' | relative_url }}" alt="CockroachDB Parallel Statement Execution" style="border:1px solid #eee;max-width:100%" />
+
+### Perceived delay in execution of barrier statements 
+
+As stated earlier, barrier statements are executed only after all parallel statements are executed. Thus it may seem as if the barrier statement is taking longer to execute, but it's in fact waiting on the parallel statements. Even then, the total time required for parallel execution of statements followed by the sequential execution of the SQL statement should be less than time required for the sequential execution of all statements. 
+
+Referring to the previous diagram, all UPDATE statements are executed before executing COMMIT. Hence it might seem as if COMMIT is taking longer to execute, but it is, in fact, waiting on the UPDATE statements to be executed.
+
+### Error message mismatch
+
+In case of parallel statement execution, it is possible to see a mismatch in the error message being displayed and the statement being executed. 
+
+Because the `RETURNING NOTHING` clause is used with parallel statements, CockroachDB cannot return the error for the parallel statement. The error message for the parallel statement is then reported on the next barrier statement. Thus you might see a mismatch with the error message to the statement being executed. 
+
+On encountering the error, CockroachDB will stop the execution of all parallel statements and [abort and retry the transaction](transactions.html#error-handling). So even an error message mismatch occurs, the result will be the same as aborting the transaction.
+
+Revisiting our sample scenario, suppose an error occurs while executing the second UPDATE statement. In sequential execution, the transaction will be aborted and the error message will be sent immediately. But in case of parallel execution, the transaction will be aborted and error message will be sent after COMMIT. The following diagram illustrates the concept of error message mismatch:
+
+<img src="{{ 'images/Parallel_Statement_Execution_Error_Mismatch.png' | relative_url }}" alt="CockroachDB Parallel Statement Execution Error Mismatch" style="border:1px solid #eee;max-width:100%" />
+
+### `RETURNING NOTHING` clause appended to dependent statements
+
+If two consecutive statements are not independent, and yet a `RETURNING NOTHING` clause is added to the statements, CockroachDB will detect the dependence and serialize the execution. You can thus use the `RETURNING NOTHING` clause with SQL statements without worrying about their dependence.
+
+Revising our sample scenario, suppose we want to create a new user on the social networking app. We need to create entries for the last name of the user, their favorite movie, and favorite song. We need to insert entries into three tables: username, movies, and songs. The transaction would be as follows:
+
+~~~ sql
+> BEGIN;
+> id = 2;
```

Comment:
no need for `> id = 2;`

Reply:
Will the hybrid case be evident without id=2?

Reply:
I think so.